### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by BradRuderman** [jsforce/jsforce#247]
Anyone experiencing and issue with this warning:

(node) warning: possible EventEmitter memory leak detected. 11 value listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at addListener (events.js:179:15)
    at Tooling.initialize (<#/ProjectPath#>/node_modules/jsforce/lib/api/tooling.js:62:36)
    at Connection.initialize (<#/ProjectPath#>/node_modules/jsforce/lib/connection.js:216:18)
    at <#/ProjectPath#>/node_modules/jsforce/lib/connection.js:1059:10
    at _fulfilled (<#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (<#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (<#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:796:13)
    at <#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:604:44
    at runSingle (<#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:137:13)
    at flush (<#/ProjectPath#>/node_modules/jsforce/node_modules/q/q.js:125:13)

**Comment by katylava** [comment]
Yes this is happening to me with jsforce 1.5.0 and node 0.12.7.

**Comment by imranansari** [comment]
any update on this one?, i have the same issue

**Comment by colestrode** [comment]
I'm seeing this too on jsforce 1.3.1 and node 0.12.7

**Comment by stomita** [comment]
There are many different cases for happening the listener limit exception to occur. Please provide the code when it happens. I recommend using the latest (1.5.x) jsforce.

**Comment by bizob2828** [comment]
Ran this with 1.5.0 on v0.12.7 and still see this

```(node) warning: possible EventEmitter memory leak detected. 11 value listeners added. Use emitter.setMaxListeners() to increase limit.
>> Trace
>>     at addListener (events.js:179:15)
>>     at Tooling.initialize (/Users/bevans/code/accounts-api/api/node_modules/jsforce/lib/api/tooling.js:63:36)
>>     at Connection.initialize (/Users/bevans/code/accounts-api/api/node_modules/jsforce/lib/connection.js:229:18)
>>     at /Users/bevans/code/accounts-api/api/node_modules/jsforce/lib/connection.js:1073:10
>>     at /Users/bevans/code/accounts-api/api/node_modules/jsforce/node_modules/promise/lib/core.js:33:15
>>     at flush (/Users/bevans/code/accounts-api/api/node_modules/jsforce/node_modules/promise/node_modules/asap/asap.js:27:13)
>>     at process._tickCallback (node.js:355:11)```

**Comment by stomita** [comment]
@bizob2828 Could you also file the code when you encountered the error ?

**Comment by bizob2828** [comment]
@stomita you mean the calls to the jsforce methods? I may have to obfuscate some of our code

**Comment by stomita** [comment]
@bizob2828 Not the actual business code but the minimum one to reproduce the same error.

**Comment by alexandremaeda** [comment]
I have the same problem here.
An idea of how my code is:

    ctrl.new = function (req, res) {
        jf.login('username@domain.com', 'password', function (loginErr, res) {
        var lead = {}
        jf.sobject("Lead").create(lead, function (err, ret) {
            if (err)
                getError(res, err, 500)
            else
                res.status(200).end()
            })
        })
    }

**Comment by born2net** [comment]
seeing same issue, any fix planned?

**Comment by stomita** [comment]
I found it occurs when the invalid instanceUrl is assigned to the connection. The fix of #197 will remove the case when assigned different instanceUrl and remove the issue refreshing tokens infinitely (and leads to event emitter memory leak).

**Comment by colestrode** [comment]
Awesome! Thanks for tracking it down :) Any chance you're going to republish with a version bump so we can pull this change in?

**Comment by stomita** [comment]
@colestrode The fix will be in 1.5.1 and released till the end of this month.

**Comment by colestrode** [comment]
Great news! Thanks again!
